core/validation/pre_transformation_validate.py

```diff
@@ -0,0 +1,148 @@
+    """
+    authorisation_checks = {}
+    sheet = workbook.get("2 - Project Admin")
+
+    if str(sheet.iloc[6][4]).strip() in TF_PLACE_NAMES_TO_ORGANISATIONS.keys():
+        authorisation_checks["Place Name"] = (
+            str(sheet.iloc[6][4]).strip(),
+            place_names,
+        )
+    if FUND_TYPE_TO_TD_OR_HS.get(sheet.iloc[5][4], sheet.iloc[5][4]) in {"TD", "HS"}:
+        authorisation_checks["Fund Type"] = (
+            FUND_TYPE_TO_TD_OR_HS.get(sheet.iloc[5][4], sheet.iloc[5][4]),
+            fund_types,
+        )
+
+    failures = check_values(authorisation_checks, "unauthorised_submission_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
+
+
+def wrong_input_validation(workbook: dict[str, pd.DataFrame], reporting_round: int):
+    """
+    Validate wrong input in the given workbook for the specified reporting round.
+
+    :param workbook: A dictionary containing sheets of the workbook.
+    :param reporting_round: An integer representing the reporting round for the validation.
+    :raises ValidationError: If wrong input validation fails, raises a ValidationError with associated failures.
+    """
+    form_version, reporting_period = GET_FORM_VERSION_AND_REPORTING_PERIOD[reporting_round]
+
+    wrong_input_checks = {}
+    sheet1 = workbook.get("1 - Start Here")
+    sheet2 = workbook.get("2 - Project Admin")
+
+    wrong_input_checks["Form Version"] = (
+        sheet1.iloc[6][1],
+        (form_version,),
+    )
+    wrong_input_checks["Reporting Period"] = (sheet1.iloc[4][1], (reporting_period,))
+    wrong_input_checks["Fund Type"] = (
+        FUND_TYPE_TO_TD_OR_HS.get(sheet2.iloc[5][4], sheet2.iloc[5][4]),
+        ("TD", "HS"),
+    )
+    wrong_input_checks["Place Name"] = (
+        str(sheet2.iloc[6][4]).strip(),
+        tuple(TF_PLACE_NAMES_TO_ORGANISATIONS.keys()),
+    )
+
+    failures = check_values(wrong_input_checks, "wrong_input_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
+
+
+def conflicting_input_validation(workbook: dict[str, pd.DataFrame]):
+    """
+    Validate conflicting input in the given workbook.
+
+    This function checks for conflicting input in the submitted workbook.
+
+    :param workbook: A dictionary containing sheets of the workbook.
+    :raises ValidationError: If conflicting input validation fails, raises a ValidationError with associated failures.
+    """
+    conflicting_input_checks = {}
+    sheet = workbook.get("2 - Project Admin")
+
+    conflicting_input_checks["Place Name vs Fund Type"] = (
+        FUND_TYPE_TO_TD_OR_HS.get(sheet.iloc[5][4]),
+        tuple(PLACE_TO_FUND_TYPE.get(str(sheet.iloc[6][4]).strip())),
+    )
+
+    failures = check_values(conflicting_input_checks, "wrong_input_failure")
+
+    if failures:
+        raise ValidationError(validation_failures=failures)
```

Comment:
This isn't very obvious what it is (or perhaps it's just Friday afternoon.. 😁)

again, it feels very specific - we need to try and refactor this so it can be used for new rounds / spreadsheets

Reply:
In effect, there's a hierarchy of pre-transformations failures. First, we check place_name and fund_type to see if the user is authorised to submit for both. Then, we check place_name and fund_type to see if the values entered into them are valid (i.e. they're part of the dropdown on the spreadsheet for both). Then, we check to see if place_name and fund_type are conflicting (i.e. if the place is Newark but the selected fund type is HS, but Newark is only associated with a TD, then the user could be authorised to submit for the place and the fund type, the place and the fund type could be valid, but they conflict with one another).

We don't want to raise multiple errors on the same cell (or else it might be confusing), so the three functions are checking if authorised > valid > conflicting, and stopping to raise error if any after each one.